Pypi package

ylzhouchris opened the pull request:
**Problem**

*1. missing files required for PyPI submission*
*2. file paths were local paths, calculation results were stored in the package folder*
*3. inconsistent outputs*
*4. needed a simple tutorial to use the package* 

**Solution**

*P1*
- added necessary files. please see commit.

*P2*
- changed to relative references based on an added parameter <data_dir> set by the user to specify the location of a project folder where all the computation will take place. 
- added a test mode that downloads a sample dataset from Github for demonstration purposes

*P3*
- adjusted all print-outs

*P4*
- see the added notebook


 
**Tests**

*Please go through the tutorial notebook*

ylzhouchris committed:
reorganize code for pypi

ylzhouchris committed:
PyPi submit - preprocess & train

Problem
- hard to retrieve data directory

Fix
- 1. add input parameter data_dir to store all the processed results, instead of storing in the code directory.
- 2. add test data to be loaded from Github

ylzhouchris committed:
re-structure folders for pypi submission

1. change package name
2. change folder structure
3. add setup.py

ylzhouchris committed:
code fixes for pypi

1. fix imports
2. load test data from github
3. adjust data directories
4. adjust for loops to process all trained models if specified
5. adjust printouts

ylzhouchris committed:
updated tutorial

add markdown scripts

jirvin16 commented on fluxgapfill/gapfill/gapfill.py:
```diff
@@ -0,0 +1,281 @@
+    Gapfill data with trained models.
+
+    Args:
+        # model_dirs (list<str>): Comma-separated list of paths to model
+        #                         directories with checkpoints. These must all
+        #                         be from the same site. sites, models,
+        #                         predictors, and predictors_paths are ignored
+        #                         if this parameter is supplied to the function.
```
If this is removed, should delete this from the docstring?

jirvin16 commented on fluxgapfill/test/test.py:
```diff
@@ -0,0 +1,296 @@
+    Args:
+        data_dir (<str>): 
+        
+        # model_dirs (list<str>): Comma-separated list of paths to model
+        #                         directories with checkpoints. These must all
+        #                         be from the same site. sites, models,
+        #                         predictors, and predictors_paths are ignored
+        #                         if this parameter is supplied to the function.
```
Same comment about this.

jirvin16 commented on fluxgapfill/test/test.py:
```diff
@@ -0,0 +1,296 @@
+        if "results" in sites:
+            sites.remove("results")
+    
+    ##
+    # model_dirs = parse_model_dirs(
+    #     model_dirs=model_dirs,
+    #     sites=sites,
+    #     models=models,
+    #     predictors=predictors,
+    #     predictors_paths=predictors_paths,
+    #     data_dir=data_dir
+    # )
```
+1

jirvin16 commented on fluxgapfill/test/test.py:
```diff
@@ -0,0 +1,296 @@
+                eval_scores_df = pd.concat(eval_scores)
+                eval_scores_df.to_csv(eval_scores_fn, index=False)
+
+    # eval_scores_df = pd.concat(eval_scores)
+    # print(f"Writing aggregated {split} metrics to {eval_scores_fn}")
+    # eval_scores_df.to_csv(eval_scores_fn, index=False)
```
Not writing aggregated metrics anymore?

jirvin16 commented on setup.py:
```diff
@@ -0,0 +1,131 @@
+# Package meta-data.
+NAME = 'fluxgapfill'
+DESCRIPTION = 'A package for machine learning driven methane gapfilling.'
+URL = 'https://github.com/stanfordmlgroup/methane-gapfill-mlt'
```
```suggestion
URL = 'https://github.com/stanfordmlgroup/methane-gapfill-ml'
```

jirvin16 reviewed (approved):
This looks great!

I didn't do too much of a deep dive into the code as much of it was copied, but please let me know if there are certain places where I should be more scrutinous.

ylzhouchris committed:
Update setup.py

fix URL typo

Co-authored-by: Jeremy Irvin <jirvin16@stanford.edu>

ylzhouchris committed:
Update test.py

update data_dir and remove model_dirs in docstring

ylzhouchris committed:
Update __version__.py

update package version to be consistent with setup.py

ylzhouchris committed:
Update gapfill.py

update docstring
- add data_dir
- remove model_dirs

ylzhouchris committed:
Update preprocess.py

update docstring
- add data_dir

ylzhouchris committed:
Update test.py

removed for loop of model_dirs, since all models will be saved in data_dir/site/predictor/models

ylzhouchris committed:
Update train.py

@jirvin16 Hi Jeremy, can you double-check the codes to detect if existing models have the same argument with the current run? Since not all parameters were given to train() while doing step-to-step gap-filling, models trained step-by-step cannot be directly used while re-running using the run_all() command raising the ValueError (Line 120). Let me know if I didn't explain clearly enough.

ylzhouchris committed:
update readme

ylzhouchris committed:
Merge branch 'pip-package' of https://github.com/stanfordmlgroup/methane-gapfill-ml into pip-package

ylzhouchris committed:
update readme and version

for pypi submission
